modeline/net/net.lisp

```diff
@@ -135,5 +148,39 @@ For the second case rescans route table every minute."
 	  (list (/ x 1e3) "k")))
     (setq dn (kbmb (car net) 0.1)
 	  up (kbmb (cadr net) 0.1))
-    (format nil "~A: ~5,2F~A/~5,2F~A " (net-device)
+    (format nil "~5,2F~A/~5,2F~A "
 	    (car dn) (cadr dn) (car up) (cadr up))))
+
+(defun fmt-ipv4 ()
+   (if *net-ipv4*
+       *net-ipv4*
+       "noip"))
+
+(defun fmt-ipv6 ()
+  (if *net-ipv6*
+      *net-ipv6*
+      "noip"))
+
+(defun net-modeline (ml)
+  (declare (ignore ml))
+  (format-expand *net-formatters-alist*
+                 *net-modeline-fmt*))
+
+(defvar *net-formatters-alist*
+  '((#\d net-device)
```

Comment:
It is a frequent mistake to think [`a`ssociation `lists`](http://www.lispworks.com/documentation/HyperSpec/Body/26_glo_a.htm#association_list) are a list of lists, but they are a list of pairs. `*NET-FORMATTERS-ALIST*` should be:

```lisp
(defvar *net-formatters-alist*
  '((#\d . net-device)
    (#\u . fmt-net-usage)
    (#\i . fmt-ipv4)
    (#\I . fmt-ipv6)))
```